route/example_option_test.go

```diff
@@ -3381,3 +3383,1041 @@ func ExampleInitializationCosts() {
+		stops,
+		vehicles,
+		route.Threads(1),
+		route.Sorter(
+			func(_ route.PartialPlan,
+				_, vehicles model.Domain,
+				_ *rand.Rand,
+			) []int {
+				orderedVehicles := vehicles.Slice()
+				// try to assign to the given vehicles in the reverse order
+				sort.SliceStable(orderedVehicles, func(i, j int) bool {
+					return orderedVehicles[i] > orderedVehicles[j]
+				})
+				return orderedVehicles
+			}),
+	)
+	if err != nil {
+		panic(err)
+	}
+	solver, err := router.Solver(store.DefaultOptions())
+	if err != nil {
+		panic(err)
+	}
+
+	// Get the last solution of the problem and print it.
+	last := solver.Last(context.Background())
+	b, err := json.MarshalIndent(last.Store, "", "  ")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(string(b))
+	// Output:
+	// {
+	//   "unassigned": [],
+	//   "vehicles": [
+	//     {
+	//       "id": "v1",
+	//       "route": [
+	//         {
+	//           "id": "Arashiyama Bamboo Forest",
+	//           "position": {
+	//             "lon": 135.672009,
+	//             "lat": 35.017209
+	//           }
+	//         }
+	//       ],
+	//       "route_duration": 0
+	//     },
+	//     {
+	//       "id": "v2",
+	//       "route": [
+	//         {
+	//           "id": "Fushimi Inari Taisha",
+	//           "position": {
+	//             "lon": 135.772695,
+	//             "lat": 34.967146
+	//           }
+	//         },
+	//         {
+	//           "id": "Kiyomizu-dera",
+	//           "position": {
+	//             "lon": 135.78506,
+	//             "lat": 34.994857
+	//           }
+	//         },
+	//         {
+	//           "id": "Gionmachi",
+	//           "position": {
+	//             "lon": 135.775682,
+	//             "lat": 35.002457
+	//           }
+	//         },
+	//         {
+	//           "id": "Kyoto Imperial Palace",
+	//           "position": {
+	//             "lon": 135.762057,
+	//             "lat": 35.025431
+	//           }
+	//         },
+	//         {
+	//           "id": "Nijō Castle",
+	//           "position": {
+	//             "lon": 135.748134,
+	//             "lat": 35.014239
+	//           }
+	//         },
+	//         {
+	//           "id": "Kinkaku-ji",
+	//           "position": {
+	//             "lon": 135.728898,
+	//             "lat": 35.039705
+	//           }
+	//         }
+	//       ],
+	//       "route_duration": 1243
+	//     }
+	//   ]
+	// }
+}
+
+// A custom type that implements Violated to fulfill the VehicleConstraint
+// interface.
+type CustomConstraint struct {
+	count int
+}
+
+// Violated the method that must be implemented to be a used as a
+// VehicleConstraint.
+func (c CustomConstraint) Violated(
+	vehicle route.PartialVehicle,
+) (route.VehicleConstraint, bool) {
+	violated := len(vehicle.Route()) > c.count
+	if violated {
+		return nil, true
+	}
+	return c, false
+}
+
+// Create routes to visit seven landmarks in Kyoto using two vehicles.
+// A custom constraint is configured.
+func ExampleCustomConstraint() {
```

Comment:
Shouldn't this one be named `Constraint` because the [option](https://github.com/nextmv-io/sdk/blob/a0e15d4bfe566023c86ae87413c57bb559905d95/route/option.go#L348) is named `Constraint`?
```suggestion
func ExampleConstraint() {
```